### ISSUE TITLE
[Bug] Node-splitting models broken?

### ISSUE THREAD

**Issue by riher** [gertvv/gemtc#58]
Hey, first of all, thank you so much for this amazing package, you're doing god's work!

One problem I've stumbled upon though, not being able to resolve it:
I can't create a node-splitting model. The error message I'm getting:
```
9.
FUN(X[[i]], ...)
8.
lapply(X = X, FUN = FUN, ...)
7.
sapply(data[, "study"], function(study) { sum(data[, "study"] == study) > 1 })
6.
remove.onearm(data.re, warn = TRUE)
5.
mtc.network(data.ab = data.ab, data.re = data.re, treatments = model[["network"]][["treatments"]], studies = studies)
4.
mtc.model.nodesplit(list(type = "nodesplit", linearModel = "random", network = structure(list(description = "Network", treatments = structure(list( id = structure(1:23, .Label = c(...
3.
do.call(fn, c(list(model), list(...)))
2.
mtc.model.call("mtc.model", model, ...)
1.
mtc.model(nma$network, linearModel = "random", n.chain = 3, likelihood = "normal", link = "identity", type = "nodesplit", t1 = "...", t2 = "...")
```
Version 0.8-7 of gemtc.
Version 1.3.959 of RStudio
Version 4.0.2 of R

I'd be very grateful for any pointers on how to get this to run.

Cheers

**Comment by riher** [comment]
I looked into it a bit more. It seems to be caused by the data (data.re) being passed as a tibble, not as a data frame.

As tidyverse is common and this is pretty obscure, applying `as.data.frame` somewhere to the data would be quite helpful.

**Comment by riher** [comment]
A similar error occurred with `mtc.anohe`. It's also fixed by this change.

**Comment by gertvv** [comment]
Hi Ricardo. Thanks for reporting and looking into this. The pull request is also much appreciated, I'll leave some feedback there.

**Comment by vovoch** [comment]
I have also found one more problem with node-splitting model.
It occurs with the code which worked with absolutely no problem before.

gemtc_data - a tibble with `study`, `treatment`, `responders`, and `sampleSize `columns.

```
mynetwork <- mtc.network(data.ab = as.data.frame(gemtc_data), description = "Data1")
nodespl <- mtc.nodesplit(mynetwork, linearModel = .x, n.adapt = 20000, n.iter = 180000, n.chain = 3, thin = 3)
res <- summary(nodespl)$p.value %>% as_tibble()
```

There are several following warnings after both fixed and random models:
```
# rjags::jags.model(file.model, data = syntax[["data"]], inits = syntax[["inits"]],  :
#  Unused variable "om.scale" in data
```

**Comment by timoman85** [comment]
Thank you for all your work on gemtc. 
I am using R 4.0.3, Rstudio 1.4.1103, gemtc 0.8-8.
I am performing a network meta-analysis and I cannot get the nodesplit function to work. 
After
>network <- mtc.network(data.ab = LOSNetmeta)
Warning message:
Setting row names on a tibble is deprecated. 
> summary(network)
$Description
[1] "MTC dataset: Network"

$`Studies per treatment`
GDT LFT RFT SFT 
 67   8  27  68 

$`Number of n-arm studies`
2-arm 
   85 

$`Studies per treatment comparison`
   t1  t2 nr
1 GDT RFT  9
2 GDT SFT 58
3 LFT RFT  8
4 RFT SFT 10
> nodesplit <- mtc.nodesplit(network, 
+                            linearModel = "random", 
+                            n.adapt = 5000, 
+                            n.iter = 100000, 
+                            thin = 10)
Error in data[, "study"] == study : 
  comparison of these types is not implemented
In addition: Warning message:
In FUN(X[[i]], ...) :
  Incompatible methods ("Ops.data.frame", "Ops.factor") for "=="

I noticed riher's comment about tidyverse and using a data frame. I tried to coerce the network data into a data frame but was unable to. 

> as.data.frame(network)
Error in as.data.frame.default(network) : 
  cannot coerce class ‘"mtc.network"’ to a data.frame
> 
Any idea of what I could do? Thank you very much for your help.
